sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/PropertiesHelper.cs

```diff
@@ -0,0 +1,76 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2019 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Xml.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace SonarAnalyzer.Helpers
+{
+    internal static class PropertiesHelper
+    {
+        internal const string AnalyzeGeneratedCodeCSharp = "sonar.cs.analyzeGeneratedCode";
+        internal const string AnalyzeGeneratedCodeVisualBasic = "sonar.vbnet.analyzeGeneratedCode";
+
+        // TODO should make this a singleton class or make sure somehow to not read every time the setting
+
+        public static bool ShouldAnalyzeGeneratedCode(this AnalyzerOptions options, string language)
+        {
+            return ReadProperty(GetSettings(options), GetProperty());
+
+            string GetProperty() =>
+                language == LanguageNames.CSharp
+                    ? AnalyzeGeneratedCodeCSharp
+                    : AnalyzeGeneratedCodeVisualBasic;
+        }
+
+        private static IEnumerable<XElement> GetSettings(AnalyzerOptions options)
+        {
+            var sonarLintAdditionalFile = options.AdditionalFiles
+                .FirstOrDefault(f => ParameterLoader.IsSonarLintXml(f.Path));
+
+            if (sonarLintAdditionalFile == null)
+            {
+                return new List<XElement>();
+            }
+
+            var xml = XDocument.Load(sonarLintAdditionalFile.Path);
```

Comment:
The code that reads from the XML file is inconsistently defensive; it guards against some conditions, but not other e.g. it won't fail if a property value isn't a boolean, but it will fail if the file isn't XML.
It would be safer to be defensive on this line too.

Reply:
Update: I see there's a malformed XML test. I don't understand why `XDocument.Load` isn't throwing with that input. I've just tried calling `XDocument.Load` with a missing file, an empty file, and the content of the malformed XML file, and it threw in every case.